1.indicator_analysis.R

```diff
@@ -86,16 +86,24 @@ analyze_first <- function(filename, geography = c("council", "datazone11"),
       data_indicator <- left_join(x=data_indicator, y=geo_lookup, c("datazone" = "datazone2011")) %>% 
         mutate(scotland = as.factor("S00000001")) # adding Scotland
       
-    } else if (geography == "council") {
+    } else if (geography == "council" & adp==FALSE) {
       geo_lookup <- readRDS(paste0(lookups, 'Geography/DataZone11_All_Geographies_Lookup.rds')) %>% 
         select(ca2019, hb2019) %>% distinct %>%  rename(ca = ca2019, hb = hb2019)
       
       ## Matching with geography lookup.
       data_indicator <- left_join(x=data_indicator, y=geo_lookup, c("ca")) %>% 
         mutate(scotland = as.factor("S00000001")) # adding Scotland
       
+    } else if (geography == "council" & adp==TRUE) {
+      geo_lookup <- readRDS(paste0(lookups, 'Geography/DataZone11_All_Geographies_Lookup.rds')) %>% 
+        select(ca2019, hb2019, adp) %>% distinct %>%  rename(ca = ca2019, hb = hb2019)
```

Comment:
There is a bit of repetition going on here that makes it difficult to see what is different between the different ifs statements. I would do a nested if statement to make it more readable, but this is my personal perference, so ignore if you want:

`
geo_lookup <- readRDS(paste0(lookups, "Geography/DataZone11_All_Geographies_Lookup.rds")) 

    # Merging data with lookup depending on geography base
    if(geography == "datazone11") {
      
      ## Matching with geography lookup.
      data_indicator <- left_join(x=data_indicator, y=geo_lookup, c("datazone" = "datazone2011")) %>% 
        mutate(scotland = as.factor("S00000001")) # adding Scotland
      
    } else if (geography == "council" ) {

      if (adp == FALSE) { #different variables required if ADP included
        geo_lookup <- geo_lookup %>% select(ca2019, hb2019) 
      } else if (adp==TRUE) {
        geo_lookup <- geo_lookup %>% select(ca2019, hb2019, adp)
      }
      
      geo_lookup <- geo_lookup %>% distinct %>%  rename(ca = ca2019, hb = hb2019)
      
      ## Matching with geography lookup.
      data_indicator <- left_join(x=data_indicator, y=geo_lookup, c("ca")) %>% 
        mutate(scotland = as.factor("S00000001")) # adding Scotland
    }
    `

Reply:
your way is better....much clearer so i've updated code